### PR TITLE
Add stats by tags command and update user docs

### LINKED ISSUE #108: As an expert user, I want to be able to see the statistics of my overall performance of a certain tag
Statistics aggregated by `tag`.

### PR DESCRIPTION
**Pull request by josiahkhoo**
Closes #108.

### CODE REVIEW

**Review by ChenXJ98 — COMMENTED** [review]
LGTM! I'll wait for Josh to look through the UG before approving but thanks! 👍

**Review by ChenXJ98 — APPROVED** [review]
UG looks good to me!

File: docs/UserGuide.md [hunk 1]
```diff
@@ -219,20 +219,51 @@ Format: `test INDEX o/OPTION`
...
 * The number of times and the percentage the user answers the question associated with the flashcard incorrectly.
 
-Format: `stats INDEX`
+1. Use the `list` command to first list all the flashcards. You can also use the `find` command to filter for a list of flashcards.
+![StatsIndexStep1](./images/StatsIndexStep1.png)
+2. Using the indices of the displayed list, enter the stats command followed by the index of the flashcard you want to reveal the statistics of. For example, if you want to display the statistics of the 2nd flashcard in the displayed list, you can enter `stats 2`.
+![StatsIndexStep2](./images/StatsIndexStep2.png)
+3. Press enter and QuickCache will display the statistics of the flashcard specified by the index.
+![StatsIndexStep3](./images/StatsIndexStep3.png)
+You have successfully displayed the statistics of a flashcard!
```
Comment (joshtyf) [inline docs/UserGuide.md:247]: Could you have a newline between the picture and this sentence?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -219,20 +219,51 @@ Format: `test INDEX o/OPTION`
...
+Statistics include:
+
+* The number of times and the percentage the user answers all flashcards containing the specified tag(s) correctly.
+* The number of times and the percentage the user answers all flashcards containing the specified tag(s) incorrectly.
+
+1. In the user input box, enter the stats command together with the tags that you want to use as the criteria. For example, if you want to display statistics for all flashcards with the tag `MCQ`, type `stats t/MCQ`.
+![StatsTagsStep1](./images/StatsTagsStep1.png)
+2. Press enter and QuickCache will display the statistics of the flashcards containing the specified tags.
+![StatsTagsStep2](./images/StatsTagsStep2.png)
+You have successfully displayed the statistics of all flashcards containing the specified tags!
```
Comment (joshtyf) [inline docs/UserGuide.md:266]: Same as above, could you have a newline between the picture and this sentence?

**Review by joshtyf — COMMENTED** [review]
Minor changes needed but let's merge it first for our consult tmr! You can fix it in another commit

**Pull request merged**